Assets/PatchKit Patcher/Scripts/AppData/Remote/RemoteMetaData.cs

```diff
@@ -15,27 +13,37 @@ public class RemoteMetaData : IRemoteMetaData
         private readonly MainApiConnection _mainApiConnection;
         private readonly KeysApiConnection _keysApiConnection;
 
-        public RemoteMetaData(string appSecret) : this(appSecret, 
-            new MainApiConnection(Settings.GetMainApiConnectionSettings()),
-            new KeysApiConnection(Settings.GetKeysApiConnectionSettings()))
-        {
-            _keysApiConnection.HttpWebRequestFactory = new UnityWebRequestFactory();
-        }
-
-        public RemoteMetaData(string appSecret, MainApiConnection mainApiConnection, KeysApiConnection keysApiConnection)
+        public RemoteMetaData(string appSecret)
         {
             Checks.ArgumentNotNullOrEmpty(appSecret, "appSecret");
-            Checks.ArgumentNotNull(mainApiConnection, "mainApiConnection");
-            Checks.ArgumentNotNull(keysApiConnection, "keysApiConnection");
 
             DebugLogger.LogConstructor();
             DebugLogger.LogVariable(appSecret, "appSecret");
 
             _appSecret = appSecret;
-            _mainApiConnection = mainApiConnection;
-            _keysApiConnection = keysApiConnection;
+            _mainApiConnection = new MainApiConnection(Settings.GetMainApiConnectionSettings());
+
+            var keysSettings = Settings.GetKeysApiConnectionSettings();
+
+            string overrideKeysUrl;
+
+            if (TryReadEnvironmentVariable("PK_PATCHER_KEYS_URL", out overrideKeysUrl))
```

Comment:
"PK_PATCHER_KEYS_URL" should be introduced as a constant. The best would be if all environment variables are read in single place, so all constants are listed there.